lms/djangoapps/courseware/access_response.py

```diff
@@ -0,0 +1,116 @@
+class AccessResponse(object):
+    """Class that represents a response from a has_access permission check."""
+    def __init__(self, has_access, access_error):
+        """
+        :param has_access (bool): if the user is granted access or not
+        :param access_error (AccessError): None if granted access or in certain cases of denied access, otherwise
+                                        contains specific information on why access was denied
+        """
+        self.has_access = has_access
+        self.access_error = access_error
+        if has_access:
+            assert access_error is None
+
+    def __nonzero__(self):
+        """Overrides bool() to correct truth value testing"""
+        return self.has_access
+
+    def __eq__(self, other):
```

Comment:
Why is it necessary to implement `eq` and `hash` methods for these classes?
I would have expected only `nonzero` and `to_json`.